packages/eslint-plugin-template/src/rules/no-duplicate-attributes.ts

```diff
@@ -0,0 +1,89 @@
+import { TmplAstElement } from '@angular/compiler';
+import {
+  createESLintRule,
+  getTemplateParserServices,
+} from '../utils/create-eslint-rule';
+import {
+  BoundAttribute,
+  BoundEvent,
+  TextAttribute,
+} from '@angular/compiler/src/render3/r3_ast';
+
+type Options = [];
+export type MessageIds = 'noDuplicateAttributes';
+export const RULE_NAME = 'no-duplicate-attributes';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that no components have duplicate input properties or output event listeners',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      noDuplicateAttributes:
+        'The attribute "{{attributeName}}" was declared multiple times',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const parserServices = getTemplateParserServices(context);
+
+    return {
+      Element({ inputs, outputs, attributes }: TmplAstElement) {
+        const duplicateInputsAndAttributes = findDuplicates([
+          ...inputs,
+          ...attributes,
+        ]);
+
+        const nonTwoWayBindingOutputs = outputs.filter((output) => {
```

Comment:
I did not implement the option for treating `<input [(ngModel)]="model" (ngModelChange)="model = $event">` as an error as suggested in https://github.com/angular-eslint/angular-eslint/issues/293#issuecomment-760316991 but it would probably be quite easy to add. As long as we decide not to make it the default it could be added in a followup PR and not be breaking

Reply:
I was just running low on time, I will add it today 😄

Reply:
@rafaelss95 can you think of a good name for the option?

Reply:
Would `allowTwoWayDataBinding` (that is `true` by _default_) be too weird? 🤔

Reply:
Works for me, added to the PR!